Mathlib/Order/Hom/Lex.lean

```diff
@@ -0,0 +1,177 @@
+/-
+Copyright (c) 2025 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios
+-/
+import Mathlib.Data.Sum.Order
+import Mathlib.Order.Hom.Set
+import Mathlib.Order.RelIso.Set
+
+/-!
+# Lexicographic order and order isomorphisms
+
+The main result in this file is the following: if `α` is a linear order and `x : α`, then `α` is
+order isomorphic to either of `Iio x ⊕ₗ Ici x` or `Iic x ⊕ₗ Ioi x`.
+-/
+
+open Set
+
+variable {α : Type*}
+
+/-! ### Relation isomorphism -/
+
+namespace RelIso
+
+variable {r : α → α → Prop} {x y : α} [IsTrans α r] [IsTrichotomous α r] [DecidableRel r]
+
+variable (r x) in
+/-- A relation is isomorphic to the lexicographic sum of elements less than `x` and elements not
+less than `x`. -/
+-- The explicit type signature stops `simp` from complaining.
```

Comment:
Where does this complaining happen? If you're able to make a mwe, can you file an issue with it?

Reply:
This complaining is addressed in #20479, I can make this a dependent PR so that this doesn't come up

Reply:
Do you mean #20475?